### PR TITLE
Get rid of renew_lease client code, in order to simplify the protocol

### PR DESCRIPTION
**Pull request by itamarst**
Fixes https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3773
Fixes https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3767

AFAICT the renew_lease Foolscap endpoint on the storage server is never actually used by the client, so removing it is pretty simple :grinning:

### CODE REVIEW

**Comment by itamarst** [comment]
Getting intermittent failures due to https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3776

File: src/allmydata/storage/server.py [hunk 1]
```diff
@@ -49,6 +49,9 @@
 NUM_RE=re.compile("^[0-9]+$")
 
 
+# Number of seconds to add to expiration time on lease renewal:
+DEFAULT_RENEWAL_TIME = 31 * 24 * 60 * 60
```
Comment (exarkun) [inline src/allmydata/storage/server.py:53]: I'm not sure it's so much a "default" as "the only time that is possible".  Although I would love for lease time to be a parameter eventually...

File: src/allmydata/storage/server.py [hunk 2]
```diff
@@ -351,26 +356,26 @@ def _iter_share_files(self, storage_index):
 
     def remote_add_lease(self, storage_index, renew_secret, cancel_secret,
                          owner_num=1):
-        start = time.time()
+        start = self._get_current_time()
         self.count("add-lease")
-        new_expire_time = time.time() + 31*24*60*60
+        new_expire_time = self._get_current_time() + 31*24*60*60
```
Comment (exarkun) [inline src/allmydata/storage/server.py:361]: Here's another copy of the calculation for the lease calculation

File: src/allmydata/storage/server.py [hunk 3]
```diff
@@ -351,26 +356,26 @@ def _iter_share_files(self, storage_index):
...
-        self.add_latency("add-lease", time.time() - start)
+        self.add_latency("add-lease", self._get_current_time() - start)
         return None
 
     def remote_renew_lease(self, storage_index, renew_secret):
-        start = time.time()
+        start = self._get_current_time()
         self.count("renew")
-        new_expire_time = time.time() + 31*24*60*60
+        new_expire_time = self._get_current_time() + 31*24*60*60
```
Comment (exarkun) [inline src/allmydata/storage/server.py:373]: Here's another copy of the calculation for the lease calculation

File: src/allmydata/storage/server.py [hunk 4]
```diff
@@ -563,7 +568,7 @@ def _make_lease_info(self, renew_secret, cancel_secret):
         :return LeaseInfo: Information for a new lease for a share.
         """
         ownerid = 1 # TODO
-        expire_time = time.time() + 31*24*60*60   # one month
+        expire_time = self._get_current_time() + 31*24*60*60   # one month
```
Comment (exarkun) [inline src/allmydata/storage/server.py:571]: Here's *another* copy of the calculation ...

File: src/allmydata/test/test_storage.py [hunk 5]
```diff
@@ -646,41 +648,42 @@ def test_seek(self):
...
-
-    def test_leases(self):
-        ss = self.create("test_leases")
+    def create_bucket(self, ss, storage_index, expected_already=0, expected_writers=5):
+        """
+        Given a StorageServer, create a bucket and return renewal and
+        cancellation secrets.
+        """
         canary = FakeCanary()
         sharenums = list(range(5))
```
Comment (exarkun) [inline src/allmydata/test/test_storage.py:657]: Having the 5 in here and the two expected values as parameters is a little awkward.  I'd at least document this as trying to create 5 shares.

File: src/allmydata/test/test_storage.py [hunk 6]
```diff
@@ -741,6 +744,27 @@ def test_leases(self):
         leases = list(ss.get_leases(b"si3"))
         self.failUnlessEqual(len(leases), 2)
 
+    def test_immutable_add_lease_renews(self):
+        """
+        Adding a lease on an already leased immutable just renews it.
```
Comment (exarkun) [inline src/allmydata/test/test_storage.py:749]: "... with the same renewal secret ..."

File: src/allmydata/test/test_storage.py [hunk 7]
```diff
@@ -1379,6 +1404,40 @@ def secrets(n):
               {0: ([], [(500, b"make me really bigger")], None)}, [])
         self.compare_leases_without_timestamps(all_leases, list(s0.get_leases()))
 
+    def test_mutable_add_lease_renews(self):
+        """
+        Adding a lease on an already leased mutable just renews it.
```
Comment (exarkun) [inline src/allmydata/test/test_storage.py:1409]: "... with the same lease renewal secret ..."

Comment (exarkun) [reply src/allmydata/test/test_storage.py:1409]: I wonder if there are any tests that exercise creating *multiple* leases ... ie, passing different renew secrets in the right place?

**Review by exarkun — APPROVED** [review]
Thanks.  I left some comments inline - all pretty minor, I think.  Address and merge (or, happy to do another pass after more changes if you want).

File: src/allmydata/test/test_storage.py (hunk 7 above)
Comment (itamarst) [reply src/allmydata/test/test_storage.py:1409]: `test_leases` might? Feel like this branch is enough for now though.

**Pull request merged**